### DISCUSSION TITLE
unofficial PS3 Controller on Ubuntu 22.04 NOT working

### DISCUSSION THREAD

**Discussion by burakyueksel** [bitcraze/discussions#576] [Q&A]
Hi,

Did anyone manage to get PS3 Controller working on Ubuntu 22.04?

Small history here:

- Using xboxdrv drivers I can see the PS3 Controller connected via USB.
- I can see it in cflient too.
- However axes and buttons are not working.
- Tested with jstest-gtk: also not working
- PS3 controllers all 4 leds are blinking red, indicating that device is not paired.
- I do not see PS3 controller in the list of bluetooth devices
- Sixpair and sixad is usually the tool for doing the bluetooth pairing. But the usual ppa is not available for Ubuntu 22.04 (ppa:falk-t-j/qtsixa).
- Hence I have compiled it from the source (https://github.com/digitalbirdo/sixpair/archive/master.zip). I can open sixad, but I still cannot pair it.

So at the end I have PS3 controller connected via USB to the Laptop with Ubuntu 22.04, controllers LEDs on the back (all 4 of them are blinking), I see the connected controller when I run the driver (see the below output), but I cannot see any change neither in jstest-gtk nor in cflient, nor when I run `jscal -c /dev/input/js0` in the terminal, when I move the axes or the buttons.

```console
burak@burak-Yoga-7-linux:~/code$ sudo xboxdrv --detach-kernel-driver
[sudo] password for burak: 
xboxdrv 0.8.8 - http://pingus.seul.org/~grumbel/xboxdrv/ 
Copyright © 2008-2011 Ingo Ruhnke <grumbel@gmail.com> 
Licensed under GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html> 
This program comes with ABSOLUTELY NO WARRANTY. 
This is free software, and you are welcome to redistribute it under certain conditions; see the file COPYING for details. 

Controller:        PLAYSTATION(R)3 Controller
Vendor/Product:    054c:0268
USB Path:          003:006
Controller Type:   Playstation 3 USB

Your Xbox/Xbox360 controller should now be available as:
  /dev/input/js0
  /dev/input/event15

Press Ctrl-C to quit, use '--silent' to suppress the event output
```
I appreciate any idea on how to continue here :).

Thanks!
Burak

**Comment by krichardsson** [comment]
I'm using a PS3 on Ubuntu and it works fine.

I use a cable, I have not tried BT.

I usually re-map my buttons (not sure how the default mapping is setup). Use the menu option "Input device/Configure device mapping", also [see the documentation](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#input-devices)

**Comment by burakyueksel** [comment]
Hi @krichardsson,

thank you for your quick answer!
It seems like for you everything works out of the box (this not my case, not even via usb cable). Are you also using Ubuntu 22.04.2 on your PC?

cfclient's "configure input device" was the first thing I had done. There I see my controller under the name Xbox Gamepad (userspace driver)

![image](https://user-images.githubusercontent.com/40430575/220851062-afdbbdb0-131d-4f83-86d0-55f096fe4090.png)

Do you see the same name? I wonder if using xbox drivers is the problem.

Then when I click on "Configure", and click on the highlighted "detect" buttons (example Roll) and move the axes (example right analog and horizontal axis): nothing happens.

I am stuck there.

**Reply by krichardsson** [reply]
I'm on Ubuntu 22.10 but I used 22.04 before and that worked as well

The only input device that is available for me is "Sony Playstaytion3 controller", not sure how it works. I assume your system is identifying your gamepad as a Xbox gamepad for some reason. @evoggy Do you have any input on this?
I found this https://github.com/bitcraze/crazyflie-clients-python/issues/388.

Does your gamepad work in other applications?

**Reply by burakyueksel** [reply]
> The only input device that is available for me is "Sony Playstaytion3 controller"

Interesting! Did you install a specific driver for it that is not xboxdrv?

My gamepad is recognized but not working in general in Ubuntu 22.04.2 (axis motion is not recognized, see my first message, I think it is a generic issue).

**Reply by burakyueksel** [reply]
I must also admit, I bought this gamepad, and then when it arrived I have realized that it does not have "Sony" label on it:

https://www.kaufland.de/product/443561936/

So I might be suffering from a fake gamepad after all.

**Reply by krichardsson** [reply]
Oh no! That might explain it. My gamepad looks like https://www.kaufland.de/product/139048261/?vid=443561936 or https://www.kaufland.de/product/77217121/?vid=443561936, which indeed is slightly different

I don't think I have installed any drivers for it

**Reply by burakyueksel** [reply]
Got it! Thank you for the links! I think I will look a bit more before ordering :).

For a reference, this driver also did not work:

https://github.com/chrippa/ds4drv

When I run it, the name of the gamepad changes in cfclient

![image](https://user-images.githubusercontent.com/40430575/221205427-49acd6fc-604b-4d68-9807-ffbc632a7700.png)

and also jstest-gtk sees it differently. But still not reaction.

Do you know other 4 analog axis gamepads, not necessarily ps3 controller?

**Comment by burakyueksel** [comment]
To do a follow-up on that:

- It seems to be that "fake" ps3 controllers (alternative to sony) are working on Ubuntu 20.04, but not on 22.04.2. See also the discussions here:
https://www.reddit.com/r/linuxquestions/comments/10o6352/connecting_ps3_controllers_to_ubuntu/
- I am not sure yet, but I think the issue stems from the fact that you cannot allow the controller (gamepad) without enabling it via Bluetooth. Usually a window pops up for it once the controller is connected via USB, but this is not there for 22.04.2.
- I am not sure if every original Sony PS3 Controller would work out of box in Ubuntu 22.04.2

So before buying more gamepads and testing them, I will fight a bit more to get it run. There are some patches in the internet.

Disclaimer: this might take some time for me. I am in parental leave and taking care of 2 little kids, so I am doing this during my "fun time", which usually does not last longer than 15 mins in a row :D

**Reply by knmcguire** [reply]
thanks for the updates! Yeah... joysticks or controllers can be a bit of a pain if they are not the standard... but keep us up to date if you find anything!

I'll update the title of your discussion though to indicate to others that you where not using the official ps3 controller.
